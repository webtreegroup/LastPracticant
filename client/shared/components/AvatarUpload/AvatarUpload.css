.upload-avatar {
    border-radius: 100%;
    cursor: pointer;

    input {
        display: none;
    }
}
