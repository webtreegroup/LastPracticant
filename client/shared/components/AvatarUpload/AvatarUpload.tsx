import React, { FC, memo } from 'react';
import { Avatar, InputProps } from '@material-ui/core';
import './AvatarUpload.css';
import { LOCAL } from 'client/shared/consts';

export interface InputAvatarProps extends InputProps {
    name: string;
    src?: string;
    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const AvatarUploadComponent: FC<InputAvatarProps> = memo(
    ({ name, src, onChange }) => (
    <label htmlFor={`upload_${name}`} className="upload-avatar">
        <input
            id={`upload_${name}`}
            name={name}
            onChange={onChange}
            type="file"
        />
        <Avatar src={src}>
            {LOCAL.AVATAR_DEFAULT}
        </Avatar>
    </label>
    ),
);

export const AvatarUpload = React.memo(AvatarUploadComponent);
