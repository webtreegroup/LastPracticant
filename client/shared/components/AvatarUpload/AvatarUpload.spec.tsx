import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { AvatarUpload } from './AvatarUpload';

afterEach(cleanup);

describe('AvatarUpload', () => {
    it('Render AvatarUpload component', () => {
        const { asFragment } = render(<AvatarUpload name="login" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Render AvatarUpload component with image', () => {
        // eslint-disable-next-line max-len
        const urlImg = 'https://ya-praktikum.tech/api/v2/resources/2fe9d2f9-f3e6-4187-9087-06875845a9a3/cf582f84-7be4-4965-aab8-b97fb7c2afe6_logotip-gruppy-alisa.jpg';
        const { asFragment } = render(<AvatarUpload src={urlImg} name="login" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
