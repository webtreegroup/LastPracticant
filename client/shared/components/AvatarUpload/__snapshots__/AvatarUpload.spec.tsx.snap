// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarUpload Render AvatarUpload component 1`] = `
<DocumentFragment>
  <label
    class="upload-avatar"
    for="upload_login"
  >
    <input
      id="upload_login"
      name="login"
      type="file"
    />
    <div
      class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      LP
    </div>
  </label>
</DocumentFragment>
`;

exports[`AvatarUpload Render AvatarUpload component with image 1`] = `
<DocumentFragment>
  <label
    class="upload-avatar"
    for="upload_login"
  >
    <input
      id="upload_login"
      name="login"
      type="file"
    />
    <div
      class="MuiAvatar-root MuiAvatar-circle"
    >
      <img
        class="MuiAvatar-img"
        src="https://ya-praktikum.tech/api/v2/resources/2fe9d2f9-f3e6-4187-9087-06875845a9a3/cf582f84-7be4-4965-aab8-b97fb7c2afe6_logotip-gruppy-alisa.jpg"
      />
    </div>
  </label>
</DocumentFragment>
`;
